relay-server/src/endpoints/health_check.rs

```diff
@@ -4,17 +4,28 @@ use axum::extract::Path;
 use axum::http::StatusCode;
 use axum::response::IntoResponse;
 use serde::Serialize;
+use tokio::time::Instant;
 
 use crate::service::ServiceState;
-use crate::services::health_check::IsHealthy;
+use crate::services::health_check::{IsHealthy, IsHealthyWrapper};
+use crate::statsd::RelayTimers;
 
 #[derive(Serialize)]
 struct Status {
     is_healthy: bool,
 }
 
 pub async fn handle(state: ServiceState, Path(kind): Path<IsHealthy>) -> impl IntoResponse {
-    match state.health_check().send(kind).await {
+    let received = Instant::now();
+
+    let wrapper = IsHealthyWrapper { kind, received };
+    let result = state.health_check().send(wrapper).await;
+
+    relay_statsd::metric!(
+        timer(RelayTimers::HealthcheckEndpointCheckDuration) = received.elapsed(),
```

Comment:
The middleware metric should be pretty much the same as this one, but emitting a new one as I'm not sure what metrics we can trust.